kratos/processes/calculate_nodal_area_process.h

```diff
@@ -59,38 +43,71 @@ typedef  ModelPart::ElementsContainerType ElementsContainerType;
 ///@name Kratos Classes
 ///@{
 
-/// Short class definition.
-/** Detail class definition.
-*/
-class CalculateNodalAreaProcess
+/**
+ * @brief This struct is used in order to identify when using the hitorical and non historical variables
+ */
+struct CalculateNodalAreaSettings
+{
+    // Defining clearer options
+    constexpr static bool SaveAsHistoricalVariable = true;
+    constexpr static bool SaveAsNonHistoricalVariable = false;
+};
+    
+/** 
+ * @class CalculateNodalAreaProcess
+ * @ingroup KratosCore 
+ * @brief Computes NODAL_AREA
+ * @details Calculate the NODAL_AREA for computing the weighted area in each node
+ * @author Riccardo Rossi
+ * @author Vicente Mataix Ferrandiz
+ */
+template<bool THistorical = true>
+class KRATOS_API(KRATOS_CORE) CalculateNodalAreaProcess
     : public Process
 {
 public:
     ///@name Type Definitions
     ///@{
 
+    /// Index type definition
+    typedef std::size_t IndexType;
+    
+    /// Size type definition
+    typedef std::size_t SizeType;
+
+    /// The definition of the node
+    typedef Node<3> NodeType;
+    
     /// Pointer definition of CalculateNodalAreaProcess
     KRATOS_CLASS_POINTER_DEFINITION(CalculateNodalAreaProcess);
 
     ///@}
     ///@name Life Cycle
     ///@{
 
-    /// Default constructor.
-    /// avg_elems ------ expected number of neighbour elements per node.,
-    /// avg_nodes ------ expected number of neighbour Nodes
-    /// the better the guess for the quantities above the less memory occupied and the fastest the algorithm
-    CalculateNodalAreaProcess(ModelPart& model_part, unsigned int domain_size)
-        : mr_model_part(model_part), mdomain_size(domain_size)
+    /**
+     * @brief Default constructor.
+     * @param rModelPart The model part to be computed
+     * @param DomainSize The size of the space, if the value is not provided will compute from the model part
+     */
+    CalculateNodalAreaProcess(
+        ModelPart& rModelPart, 
+        const SizeType DomainSize = 0
+        ): mrModelPart(rModelPart),
+           mDomainSize(DomainSize)
     {
+        // In case is not provided we will take from the model part
+        if (mDomainSize == 0) {
+            KRATOS_ERROR_IF_NOT(rModelPart.GetProcessInfo().Has(DOMAIN_SIZE)) << "\"DOMAIN_SIZE\" has to be specified in the ProcessInfo if using this constructor!" << std::endl;
```

Comment:
but now the error message does not fit if you don't have a second constructor
```suggestion
            KRATOS_ERROR_IF_NOT(rModelPart.GetProcessInfo().Has(DOMAIN_SIZE)) << "\"DOMAIN_SIZE\" has to be specified in the ProcessInfo" << std::endl;
```